llvm-runtimes/libgcc/libgcc-19.1.4.ebuild

```
@@ -0,0 +1,47 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit crossdev llvm.org toolchain-funcs
+
+DESCRIPTION="Compiler runtime library for clang, compatible with libgcc_s"
+HOMEPAGE="https://llvm.org/"
+
+LICENSE="Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~mips ~ppc64 ~riscv ~x86 ~amd64-linux ~arm64-macos ~ppc-macos ~x64-macos"
+
+DEPEND="
+	llvm-runtimes/compiler-rt:${LLVM_MAJOR}
+	=llvm-runtimes/libunwind-${PV}
+	!!sys-devel/gcc
+"
+BDEPEND="
+	llvm-core/clang:${LLVM_MAJOR}
+"
+
+LLVM_COMPONENTS=( llvm-libgcc )
+llvm.org_set_globals
+
+src_compile() {
+	$(tc-getCC) -E -xc \
+		"${WORKDIR}/llvm-libgcc/gcc_s.ver.in" \
+		-o gcc_s.ver || die
+	$(tc-getCC) -nostdlib \
+		-Wl,--version-script,gcc_s.ver \
+		-Wl,--whole-archive \
+		"${EPREFIX}/usr/lib/clang/${LLVM_MAJOR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" \
+		"${EPREFIX}/usr/lib/libunwind.a" \
+		-Wl,-soname,libgcc_s.so.1.0 \
+		-lc -shared \
+		-o libgcc_s.so.1.0 || die
+}
+
+src_install() {
+	dolib.so libgcc_s.so.1.0
+	dosym "clang/${LLVM_MAJOR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" /usr/lib/libgcc.a
```

```suggestion
	dosym "../lib/clang/${LLVM_MAJOR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" /usr/$(get_libdir)/libgcc.a
```

Also reuse the result of the `clang` call above to get the right path.

---

That works great, but triggers a QA notice:

```
 * QA Notice: Found an absolute symlink in a library directory:
 *            /usr/lib/libgcc.a -> /usr/lib/llvm/19/bin/../../../../lib/clang/19/lib/linux/libclang_rt.builtins-x86_64.a
 *            It should be a relative symlink if in the same directory
 *            or a linker script if it crosses the /usr boundary.
```

Not sure what to do about it. Is there a way to ignore it in a way users won't see it?

---

Does this work?
```suggestion
	dosym -r "/usr/lib/clang/${LLVM_MAJOR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" /usr/$(get_libdir)/libgcc.a
```
This should generate a relative symlink instead.

---

```shell
dosym -r "${rtlib}" "/usr/${libdir}/libgcc.a"
```

does work indeed

---

In the end, I don't need this. Since I'm building a separate copy if compiler-rt, I'm simply installing this file.